state/protocol/badger/snapshot_test.go

```diff
@@ -304,22 +310,22 @@ func TestSealingSegment(t *testing.T) {
 			block1 := unittest.BlockWithParentFixture(head)
 			buildFinalizedBlock(t, state, block1)
 
+			receipt1, seal1 := unittest.ReceiptAndSealForBlock(block1)
+
 			parent := block1
 			// build a large chain of intermediary blocks
 			for i := 0; i < 100; i++ {
 				next := unittest.BlockWithParentFixture(parent.Header)
+				next.SetPayload(unittest.PayloadFixture(unittest.WithReceipts(receipt1)))
```

Comment:
repetitions of the same receipt in one fork would be a protocol violation. I would suggest to test with valid cases, and only include the receipt once:


```suggestion
				if i == 0 {
					// Repetitions of the same receipt in one fork would be a protocol violation.
					// Hence, we include the result only once in the direct child of B1. 
					next.SetPayload(unittest.PayloadFixture(unittest.WithReceipts(receipt1)))
				}
```